code/game/objects/structures/tables_racks.dm

```diff
@@ -219,10 +224,14 @@
 			return TRUE
 
 
-/obj/structure/table/CheckExit(atom/movable/mover, direction)
-	. = ..()
-	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
-		return TRUE
+/obj/structure/table/proc/on_try_exit(datum/source, atom/movable/mover, direction, list/knownblockers)
+	SIGNAL_HANDLER
+	if(CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
+		return NONE
+	if(!density || !(flags_atom & ON_BORDER) || !(direction & dir) || (mover.status_flags & INCORPOREAL))
```

Comment:
ditto on the dir check, and the density, are you sure?

Reply:
legacy code for onborder, this is maintiaining parity